File: apps/android/app/src/main/kotlin/jiamin/chen/orangecloud/data/model/URLScannerModels.kt
package jiamin.chen.orangecloud.data.model

import kotlinx.serialization.Serializable

/**
 * Cloudflare URL Scanner（**v2**，别用 v1）。
 * 扫描是异步的：提交后返回 uuid，结果需轮询，未就绪时接口 404。
 */
@Serializable
data class URLScanSubmit(val url: String)

@Serializable
data class URLScanAccepted(
    val uuid: String? = null,
    val api: String? = null,
    val message: String? = null,
    val result: String? = null,
)

@Serializable
data class URLScanResult(
    val task: URLScanTask? = null,
    val page: URLScanPage? = null,
    val verdicts: URLScanVerdicts? = null,
)

@Serializable
data class URLScanTask(
    val uuid: String? = null,
    val url: String? = null,
    val effectiveUrl: String? = null,
    val status: String? = null,
)

@Serializable
data class URLScanPage(
    val url: String? = null,
    val domain: String? = null,
    val ip: String? = null,
    val country: String? = null,
    val asn: String? = null,
    val asnname: String? = null,
    val server: String? = null,
    val statusCode: Int? = null,
)

@Serializable
data class URLScanVerdicts(val overall: URLScanVerdict? = null)

@Serializable
data class URLScanVerdict(
    val malicious: Boolean? = null,
    val categories: List<URLScanCategory>? = null,
)

@Serializable
data class URLScanCategory(val id: Int? = null, val name: String? = null)

File: apps/ios/Orange Cloud/Orange Cloud/Models/URLScannerModels.swift
//
//  URLScannerModels.swift
//  Orange Cloud
//
//  Cloudflare URL Scanner（v2）。提交 URL 做安全扫描并看报告。
//  POST /accounts/{account_id}/urlscanner/v2/scan
//  GET  /accounts/{account_id}/urlscanner/v2/result/{scan_id}
//  GET  /accounts/{account_id}/urlscanner/v2/screenshots/{scan_id}.png
//
//  ⚠️ 只用 v2，不要用 v1（v1 是旧版，字段与路径都不同）。
//  扫描是异步的：提交后返回 uuid，结果要轮询，未就绪时 result 返回 404/202。
//

import Foundation

nonisolated struct URLScanSubmit: Codable, Sendable {
    let url: String
}

/// 提交成功后的回执
nonisolated struct URLScanAccepted: Codable, Sendable {
    let uuid:    String?
    let api:     String?
    let message: String?
    let result:  String?
}

/// 扫描报告。字段很多，这里只取移动端会展示的那部分。
nonisolated struct URLScanResult: Codable, Sendable {
    let task:     URLScanTask?
    let page:     URLScanPage?
    let verdicts: URLScanVerdicts?
    let stats:    URLScanStats?
}

nonisolated struct URLScanTask: Codable, Sendable {
    let uuid:      String?
    let url:       String?
    let effectiveUrl: String?
    let status:    String?
    let time:      String?

    enum CodingKeys: String, CodingKey {
        case uuid, url, status, time
        case effectiveUrl = "effectiveUrl"
    }
}

nonisolated struct URLScanPage: Codable, Sendable {
    let url:     String?
    let domain:  String?
    let ip:      String?
    let country: String?
    let asn:     String?
    let asnname: String?
    let server:  String?
    let statusCode: Int?

    enum CodingKeys: String, CodingKey {
        case url, domain, ip, country, asn, asnname, server
        case statusCode = "statusCode"
    }
}

nonisolated struct URLScanVerdicts: Codable, Sendable {
    let overall: URLScanVerdict?
}

nonisolated struct URLScanVerdict: Codable, Sendable {
    let malicious:  Bool?
    let categories: [URLScanCategory]?
    let phishing:   [String]?
}

nonisolated struct URLScanCategory: Codable, Sendable {
    let id:   Int?
    let name: String?
}

nonisolated struct URLScanStats: Codable, Sendable {
    let uniqCountries: Int?
    let dataLength:    Int?
    let requests:      Int?
}
